### ISSUE TITLE
Exclude exceptions of certain types

### ISSUE THREAD

**Issue by ghost** [coderberry/airbrake-grails#34]
### What?

Add a configuration param that can be used to skip notice-sending for certain types of exceptions. For example, imagine I don't want to send a notice for all exceptions of type IllegalArgumentException or IllegalStateException

This could be configured using a list of exception class names:

``` groovy
grails.plugins.airbrake.excludes = ['java.lang.IllegalArgumentException', 'java.lang.IllegalStateException']
```

or alternatively, one or more exception class name patterns:

``` groovy
grails.plugins.airbrake.excludes = ['java.lang.Illegal.*Exception']
```

I considered a list of exception classes instead of a list of exception class names, but decided against the former approach because it would require the exception to be on the classpath at compile-time. 
### Why?

In my application I have a global error handler configured in `UrlMappings.groovy` 

``` groovy
"500"(controller: "errors", action: "serverError")
```

which does the following (among other things)

``` groovy
class ErrorsController {
  def serverError = {
    log.error 'unhandled exception', request.exception
  }
}
```

However, this causes the exception to be logged twice in Errbit
1. Once with the exceptions actual type and place of origin
2. Once with an exception type of GrailsWrappedRuntimeException. The origin of this second occurrence is the `log.error` statement in the global error handler

So in my case I want to use this config param to exclude these duplicates:

``` groovy
grails.plugins.airbrake.excludes = ['org.codehaus.groovy.grails.web.errors.GrailsWrappedRuntimeException']
```

I can imagine there might be other cases where this feature might be useful, e.g. a 3rd party library is throwing an exception that you don't care about (and can't change the code to prevent it).

**Comment by jonspalmer** [comment]
As an aside it seems like a really strange thing to log an error inside your 500 controller. Not only with that result in duplicate Airbrake notifications but you're also going to log every exception twice in your log files.

That said it seems reasonable to add this configuration. The Ruby notifier has something similar maybe we can be inspired by that. Happy to accept a pull request that adds a simple configuration option for filtering the exceptions.

**Comment by ghost** [comment]
> As an aside it seems like a really strange thing to log an error inside your 500 controller. Not only with that result in duplicate Airbrake notifications but you're also going to log every exception twice in your log files.

Are you sure about this? If I have a file appender configured to log to `myapp.log`, the contents of the file will only show whatever data is recorded by the logging statements. In other words unhandled exceptions are not automatically sent to this appender (though they are sent to the `stacktrace.log` appender). But if I want a single file which shows all my logging data and unhandled exceptions, I don't know how to achieve this other than by logging them from the global error handler.

Anyway, to get back to the main topic, I've implemented this as described above, and will send the pull request on Monday

**Comment by jonspalmer** [comment]
released as part or 0.9.3
